src/components/views/messages/TextualBody.tsx

```diff
@@ -503,6 +505,29 @@ export default class TextualBody extends React.Component<IBodyProps, IState> {
         );
     }
 
+    /**
+     * Render a marker informing the user that, while they can see the message,
+     * it is hidden for other users.
+     */
+    private renderHiddenMessageMarker() {
+        let text;
+        const visibility = this.props.mxEvent.messageVisibility();
+        switch (visibility.visible) {
+            case true:
+                throw new Error("renderHiddenMessageMarker should only be applied to hidden messages");
+            case false:
+                if (visibility.reason) {
+                    text = _t("Message pending moderation: %(reason)s", { reason: visibility.reason });
+                } else {
+                    text = _t("Message pending moderation");
+                }
+                break;
+        }
+        return (
+            <span className="mx_EventTile_hidden">{ `(${text})` }</span>
+        );
+    }
```

Comment:
It looks like very similar to `src/components/views/messages/HiddenBody.tsx`.
Could that other component be used in this instance or are they subtlely different?

Reply:
Well, `HiddenBody` hides the body entirely (like `RedactedBody`), while `TextualBody` displays it with a warning. I'm sure that there is a way to join them but I'm not sure it would make the code any simpler.